tests/ui/vec_box_sized.rs

```diff
@@ -1,17 +1,36 @@
-struct SizedStruct {
-    _a: i32,
-}
+// run-rustfix
 
-struct UnsizedStruct {
-    _a: [i32],
-}
+#![allow(dead_code)]
+
+struct SizedStruct(i32);
+struct UnsizedStruct([i32]);
+
+/// The following should trigger the lint
+mod should_trigger {
+    use super::SizedStruct;
 
-struct StructWithVecBox {
-    sized_type: Vec<Box<SizedStruct>>,
+    struct StructWithVecBox {
+        sized_type: Vec<Box<SizedStruct>>,
+    }
+
+    struct A(Vec<Box<SizedStruct>>);
+    struct B(Vec<Vec<Box<(u32)>>>);
 }
 
-struct StructWithVecBoxButItsUnsized {
-    unsized_type: Vec<Box<UnsizedStruct>>,
+/// The following should not trigger the lint
+mod should_not_trigger {
+    use super::UnsizedStruct;
+
+    struct C(Vec<Box<UnsizedStruct>>);
+
+    struct StructWithVecBoxButItsUnsized {
+        unsized_type: Vec<Box<UnsizedStruct>>,
+    }
+
+    struct TraitVec<T: ?Sized> {
+        // Regression test for #2720. This was causing an ICE.
```

Comment:
woops, also a wrong issue number. Will fix this and the merge conflict later today.